### ISSUE TITLE
Question about the pipeline.

### ISSUE THREAD

**Issue by staplesinLA** [HarryVolek/PyTorch_Speaker_Verification#6]
Thanks for sharing your great work. As a newer to to speaker verification, I have two question here.
First, what's the purpose of `perm` and `unperm` in training script.
Second, since the original data matrix is disordered by the perm index,  so how the loss is calculated between different speakers?

Thanks a lot!

**Comment by staplesinLA** [comment]
Oh, It seems that the pair operations of `perm` and `unperm` reconstruct the original utterance order.
So, the remaining question is: why we should randomize the utterance samples?  Because those utterances are still in a same batch.

**Comment by HarryVolek** [comment]
I believe if the structure of the batch input contains information about which speaker is associated with which utterance the algorithm will fit itself on the batch positioning of the batch input instead of the contents of the recording. So I permutate and unpermutate after retrieving the loss function output.
